test/e2e/platform/platform_test.go

```diff
@@ -37,65 +39,248 @@ import (
+				Expect(out).Should(ContainSubstring("Running"))
+			})
+
+			It("IPAM", func() {
+				ipam := &platformv1.IPAM{
+					Spec: platformv1.IPAMSpec{
+						ClusterName: clsName,
+					},
+				}
+				ipam, err := tkeClient.PlatformV1().IPAMs().Create(context.Background(), ipam, metav1.CreateOptions{})
+				Expect(err).Should(BeNil())
+
+				Eventually(func() error {
+					addon, err := tkeClient.PlatformV1().IPAMs().Get(context.Background(), ipam.Name, metav1.GetOptions{})
+					if err != nil {
+						return err
+					}
+					if addon.Status.Phase != "Running" {
+						return errors.New(addon.Name + " Phase: " + string(addon.Status.Phase) + ", Reason: " + addon.Status.Reason)
+					}
+					return nil
+				}, 10*time.Minute, 10*time.Second).Should(Succeed())
+
+				out, _ := runCmd("kubectl describe ipam " + ipam.Name + " -n kube-system --kubeconfig " + tkeKubeConfigFile + " | grep Phase")
+				Expect(out).Should(ContainSubstring("Running"))
+			})
+
+			It("CronHPA", func() {
+				cronHPA := &platformv1.CronHPA{
+					Spec: platformv1.CronHPASpec{
+						ClusterName: clsName,
+					},
+				}
+				cronHPA, err := tkeClient.PlatformV1().CronHPAs().Create(context.Background(), cronHPA, metav1.CreateOptions{})
+				Expect(err).Should(BeNil())
+
+				Eventually(func() error {
+					addon, err := tkeClient.PlatformV1().CronHPAs().Get(context.Background(), cronHPA.Name, metav1.GetOptions{})
+					if err != nil {
+						return err
+					}
+					if addon.Status.Phase != "Running" {
+						return errors.New(addon.Name + " Phase: " + string(addon.Status.Phase) + ", Reason: " + addon.Status.Reason)
+					}
+					return nil
+				}, 10*time.Minute, 10*time.Second).Should(Succeed())
+
+				out, _ := runCmd("kubectl describe cronhpa " + cronHPA.Name + " -n kube-system --kubeconfig " + tkeKubeConfigFile + " | grep Phase")
+				Expect(out).Should(ContainSubstring("Running"))
+			})
+		})
+
+		It("Delete Baremetal cluster", func() {
+			Expect(deleteCluster(clsName)).Should(Succeed())
+
+			out, _ := runCmd("kubectl get clusters --kubeconfig " + tkeKubeConfigFile)
+			Expect(out).ShouldNot(ContainSubstring(clsName))
+		})
 	})
 
-	It("complete cluster lifecycle", func() {
-		By("create baremetal cluster")
-		restconf, err := t.GetKubeConfig()
-		Expect(err).To(BeNil())
-		client := tkeclientset.NewForConfigOrDie(restconf)
-		err = createBaremetalCluster(context.Background(), client, masterNodes, workerNodes)
-		Expect(err).To(BeNil())
+	Context("Imported cluster", func() {
+		var cluster *platformv1.Cluster
+		var credential platformv1.ClusterCredential
+
+		BeforeEach(func() {
+			if cluster == nil {
+				// Prepare a cluster to be imported
+				cluster, err = createCluster(1)
+				Expect(err).To(BeNil())
+
+				// Get the credential of cluster
+				credentials, err := tkeClient.PlatformV1().ClusterCredentials().List(context.Background(), metav1.ListOptions{
```

Comment:
why not use `Get` api instead of `List`？

Reply:
Let me check if we can get the credential name or not from cluster. If yes, I'll update it to use Get instead